ueberdosis/tiptap
# fix(vue-3): late-registering plugins

## Changes Overview
It seems that `y-prosemirror` 1.2.4 or 1.2.5 breaks late registering plugins, specifically in Vue 3. This might be related to a point Kevin mentioned [here](https://github.com/yjs/y-prosemirror/issues/155#issuecomment-2181395723).

The issue arises because `this.view.updateState(state)` doesn't immediately update `this.view.state`. This delay means the reconfigured state in `registerPlugin()` doesn't include newly registered plugins. 

## Implementation Approach
To address this, I've adjusted the code to use the state returned by `this.state.reconfigure` directly.

## Testing Done
I tested the changes, added Vue and React demos of a collaboration setup in combination with bubble and floating menu, and added tests.

## Verification Steps
Open the `CollaborationWithMenus` Vue demo and check if collaboration and menus are working properly.

## Checklist

## Related Issues
Closes #5259
